components/pagination/src/Pagination.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FC, createElement as h } from 'react';
+import { StandardProps, classBuilder } from '@not-govuk/component-helpers';
+
+import '../assets/Pagination.scss';
+
+export type PaginationProps = StandardProps & {
+  results: number
+  resultsPerPage: number
+  page: number
+};
+
+export const Pagination: FC<PaginationProps> = ({
+  children,
+  classBlock,
+  classModifiers,
+  className,
+  results,
+  resultsPerPage,
+  page,
+  ...attrs
+}) => {
+  const classes = classBuilder(
+    'hods-pagination',
+    classBlock,
+    classModifiers,
+    className
+  );
+
+  page = Number(page); // in case user passes in a string
+  results = Number(results);
+  resultsPerPage = Number(resultsPerPage);
+
+  const resultsFrom = (page - 1) * resultsPerPage + 1;
+  const maxResults = page * resultsPerPage;
+  const maxPages = results / resultsPerPage + 1;
+  const resultsTo = (
+    results < maxResults
+    ? results
+    : maxResults
+  );
+
+  const pages = [page - 2, page - 1, page, page + 1, page + 2];
+  const it = [0,1,2,3,4];
+  const hl = ["?page=" + pages[0], "?page=" + pages[1], "",
+    "?page=" + pages[3], "?page=" + pages[4]];
+
+  return (
+    <div className={classes()}>
+      <div className={classes('summary')}>Showing {resultsFrom} - {resultsTo} of {results} results</div>
+      <ul className={classes('list-items')}>
+        <li className={classes('item')} id="prevButton">
+          <a className={classes('link')} href={page>1 ? hl[1] : "" }>
+            <span aria-hidden="true" role="presentation">&laquo;</span> Previous
+          </a>
+        </li>
+        {it.map(i => (
+          <li className={classes('item')}>
+            {pages[i] < 1 || pages[i] > maxPages ? null : <a className={classes(pages[i] === page ? 'link--current' : 'link')} href={hl[i]}>{pages[i]}</a>}
```

Comment:
```js
classes(pages[i] === page ? 'link--current' : 'link')
```
Should be:
```js
classes('link', pages[i] === page ? 'current' : undefined)
```
Or possibly:
```js
classes('link', pages[i] === page && 'current')
```
(Can't remember how it handles a `false` as opposed to `undefined`.)

But basically you give it the modifiers and the library handles the BEM syntax for you.

Reply:
The bottom code snippet doesn't work. i get an error from Storybook when i try to use it. However `classes('link', pages[i] === page ? 'current' : undefined)` does work and is now incorporated into the code.